src/main/java/ru/skypro/homework/repository/CommentRepository.java

```diff
@@ -16,9 +16,6 @@ public interface CommentRepository extends JpaRepository<Comment, Integer> {
 
     List<Comment> findAllByAd_IdOrderByCreatedAtDesc(Integer id);
 
-//    @Query(value = "SELECT c.author FROM Comment c WHERE c.id = :idComment")
-//    Optional<User> findAuthorComment(@Param("idComment")Integer idComment);
-
-    @Query(value = "SELECT (SELECT true FROM c.author u WHERE u.id = :idAuthor) FROM Comment c WHERE c.id = :idComment")
-    boolean checkAuthorComment(@Param("idComment") Integer idComment, @Param("idAuthor") Integer idAuthor);
+    @Query(value = "SELECT true FROM Comment c WHERE c.id = :idComment AND c.author = :idAuthor")
```

Comment:
```suggestion
    @Query(value = "SELECT true FROM Comment c WHERE c.id = :idComment AND c.author.id = :idAuthor")
```
ошибка - будет исключение